plugins/samples/ab_testing/plugin.cc

```diff
@@ -0,0 +1,66 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START serviceextensions_plugin_ab_testing]
+#include <boost/url/parse.hpp>
+#include <boost/url/url.hpp>
+
+#include "absl/strings/match.h"
+#include "absl/strings/str_cat.h"
+#include "proxy_wasm_intrinsics.h"
+
+constexpr std::string_view a_path = "/v1/";
+constexpr std::string_view b_path = "/v2/";
+constexpr int percentile = 50;
+
+class MyHttpContext : public Context {
+ public:
+  explicit MyHttpContext(uint32_t id, RootContext* root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders(uint32_t headers,
+                                       bool end_of_stream) override {
+    const auto path = getRequestHeader(":path")->view();
+
+    // Checks if the current request is eligible to be served by v2 file.
+    //
+    // The decision is made by hashing the userID into an integer
+    // value between 0 and 99, then comparing the hash to a predefined
+    // percentile. If the hash value is less than or equal to the percentile,
+    // the request is served by the v2 file. Otherwise, it is served by the
+    // original file.
+    const auto usr = extractUserFromPath(path);
```

Comment:
Can you rename the variable to `user`? This goes against the [Google C++ Style Guide](https://github.com/GoogleCloudPlatform/service-extensions/pull/121/files#diff-f0449c7ae73157f77ebb17a0a75ff1f2dd7030e92fd68bb349ff05dc35d68dd3R42): "Do not abbreviate by deleting letters within a word"

Reply:
Done